RIRImageResizing/Classes/NSObject/RIRResizeImageOperation.m

```diff
@@ -0,0 +1,189 @@
+    
+    if (self = [super init])
+    {
+        _resizeParameters = resizeParameters;
+        _image = image;
+    }
+    
+    return self;
+}
+
+#pragma mark - scale
+-(nullable UIImage*)scaleToSize:(CGSize)newSize usingMode:(UIImage_RIRResizing_ResizeMode)resizeMode
+{
+    switch (resizeMode)
+    {
+        case UIImage_RIRResizing_ResizeMode_aspectFit:
+            return [self scaleToFitSize:newSize];
+            break;
+            
+        case UIImage_RIRResizing_ResizeMode_aspectFill:
+            return [self scaleToCoverSize:newSize];
+            break;
+            
+        case UIImage_RIRResizing_ResizeMode_scaleToFill:
+            return [self scaleToFillSize:newSize];
+            break;
+            
+        default:
+            break;
+    }
+    
+    NSAssert(false, @"unhandled %li",(long)resizeMode);
+    return nil;
+}
+
+-(nonnull UIImage*)scaleToFitSize:(CGSize)newSize
+{
+    /// Keep aspect ratio
+    size_t destWidth, destHeight;
+    if (self.image.size.width > self.image.size.height)
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    else
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    if (destWidth > newSize.width)
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    if (destHeight > newSize.height)
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    
+    return [self scaleToFillSize:(CGSize){
+        .width		= destWidth,
+        .height		= destHeight,
+    }];
+}
+
+-(nonnull UIImage*)scaleToFillSize:(CGSize)newSize
+{
+    UIGraphicsBeginImageContextWithOptions(newSize, NO, self.resizeParameters.scale);
+    
+    [self.image drawInRect:CGRectMake(0,0,newSize.width,newSize.height)];
+    UIImage* newImage = UIGraphicsGetImageFromCurrentImageContext();
+    
+    UIGraphicsEndImageContext();
+    
+    return newImage;
+}
+
+-(nonnull UIImage*)scaleToCoverSize:(CGSize)newSize
+{
+    size_t destWidth, destHeight;
+    CGFloat const widthRatio = newSize.width / self.image.size.width;
+    CGFloat const heightRatio = newSize.height / self.image.size.height;
+    /// Keep aspect ratio
+    if (heightRatio > widthRatio)
+    {
+        destHeight = (size_t)newSize.height;
+        destWidth = (size_t)(self.image.size.width * newSize.height / self.image.size.height);
+    }
+    else
+    {
+        destWidth = (size_t)newSize.width;
+        destHeight = (size_t)(self.image.size.height * newSize.width / self.image.size.width);
+    }
+    
+    return [self scaleToFillSize:CGSizeMake(destWidth, destHeight)];
+}
+
+#pragma mark - resizedImage
+@synthesize resizedImage;
```

Comment:
It's conventional to name your instance variables as underscore versions of the getter/setter names. Updates this to `@synthesize resizedImage = _resizedImage;` and update the getter method accordingly.